silc/md/collective_variables.py

```diff
@@ -127,3 +128,43 @@ def com_angle(r1, r2, r3):
     r = r3 - r2
     return np.arctan2(linalg.norm(np.cross(q, r)), np.dot(q, r))
 
+
+class DistancePBC(TwoPointCV):
+    """
+    Parameters
+    ----------
+
+    indcies:
+        This is a list of two sets of indices. Each sets corresponds to a (part of a) molecule.
+
+    box:
+        This is a list of three floats, corresponding to simulation box dimension (assuming cuboid box),
+        i.e. [Lx, Ly, Lz]
+
+    Example
+    -------
+
+    indices1 = [0, 1, 2, 3, 4, 5]    # atom indices of molecule 1
+    indices2 = [10, 11, 12, 13, 14, 15]    # atom indices of molecule 2
+    box = [10., 10., 10.]    # simulation box dimension
+    cv = DistancePBC([indices1, indices2], box)
+    """
+
+    def __init__(self, indices, box=[1., 1., 1.,]):
+        super().__init__(indices)
+        self.box = np.asarray(box)
+
+    @property
+    def function(self):
+        return lambda r1, r2: distance_pbc(r1, r2, self.box)
+
+
+def periodic(distance, box):
+    return np.mod(distance + box * 0.5, box) - 0.5 * box
+
+
+def distance_pbc(r1, r2, box):
+    r1 = barycenter(r1)
+    r2 = barycenter(r2)
+    dr = r1 - r2
+    return linalg.norm(periodic(dr, box))
```

Comment:
```suggestion
def wrap(distance, box):
    return np.mod(distance + box * 0.5, box) - 0.5 * box


def distance_pbc(r1, r2, box):
    r1 = barycenter(r1)
    r2 = barycenter(r2)
    dr = r1 - r2
    return linalg.norm(wrap(dr, box))

```
Maybe wrap is a better name here.